File: apps/Camera-Vitals/Android/app/src/main/java/ai/zetic/demo/cameravitals/AppConfig.kt
package ai.zetic.demo.cameravitals

/** Central configuration & tuning constants for the rPPG pipeline (mirrors iOS AppConfig). */
object AppConfig {
    // Melange model (deployed as version 2 — the low-memory 31-frame window).
    const val PERSONAL_KEY = "YOUR_MLANGE_KEY"
    const val MODEL_NAME = "realtonypark/EfficientPhys-rPPG_camera_vitals"
    const val MODEL_VERSION = 2

    // Model I/O contract.
    const val IMG_SIZE = 72
    const val CHANNELS = 3
    const val CHUNK = 30        // model output length (rPPG samples per inference)
    const val FRAMES_IN = 31    // model input frames (chunk + 1; model does diff internally)

    // Capture / cadence. stride == chunk → contiguous windows that stitch seamlessly.
    const val FPS = 30.0
    const val STRIDE = 30

    // Rolling analysis buffer of stitched rPPG samples used for the HR FFT.
    const val ANALYSIS_SAMPLES = 300       // 10 s @ 30 fps
    const val MIN_ANALYSIS_SAMPLES = 150   // 5 s before the first heart-rate reading

    // Heart-rate band.
    const val HR_BAND_LOW = 0.75   // Hz (45 bpm)
    const val HR_BAND_HIGH = 2.5   // Hz (150 bpm)
    const val MIN_BPM = 40.0
    const val MAX_BPM = 180.0

    // Quality gating.
    const val DISPLAY_QUALITY_FLOOR = 0.15
    const val LOW_LUMA_THRESHOLD = 40f

    // Guided measurement.
    const val MEASURE_DURATION_SEC = 30.0

    val frameFloatCount: Int get() = CHANNELS * IMG_SIZE * IMG_SIZE   // 15552
}

File: apps/Camera-Vitals/iOS/CameraVitals/App/AppConfig.swift
import CoreGraphics
import Foundation

/// Central configuration & tuning constants for the rPPG pipeline.
enum AppConfig {
    // Melange model. Re-upload the low-memory bundle (melange/out_w30) to this model;
    // version `nil` always pulls the LATEST version, so no code change is needed after upload.
    static let personalKey = "YOUR_MLANGE_KEY"
    static let modelName = "realtonypark/EfficientPhys-rPPG_camera_vitals"
    static let modelVersion: Int? = nil   // nil = latest

    // Model I/O contract (small window to keep the per-inference memory peak low).
    // Each inference processes `framesIn` frames and emits `chunk` rPPG samples, which are
    // STITCHED into a rolling buffer for analysis (see analysisSamples).
    static let imgSize = 72
    static let channels = 3
    static let chunk = 30         // model output length (rPPG samples per inference)
    static let framesIn = 31      // model input frames (chunk + 1; model does torch.diff internally)

    // Capture / cadence. stride == chunk → consecutive windows are contiguous (1-frame overlap),
    // so their outputs stitch into a seamless waveform.
    static let fps: Double = 30
    static let stride = 30        // run inference every 30 new frames (~1 Hz)

    // Rolling analysis buffer of stitched rPPG samples used for the HR FFT.
    static let analysisSamples = 300       // 10 s @ 30 fps
    static let minAnalysisSamples = 150    // 5 s before the first heart-rate reading

    // Heart-rate band.
    static let hrBandLow = 0.75   // Hz  (45 bpm)
    static let hrBandHigh = 2.5   // Hz  (150 bpm)
    static let minBPM = 40.0
    static let maxBPM = 180.0

    // Quality gating.
    static let displayQualityFloor = 0.15   // below this, don't refresh the shown BPM
    static let lowLumaThreshold: Float = 40  // mean ROI luma (0-255) below this => "more light"

    // Guided measurement.
    static let measureDuration: TimeInterval = 30

    static var frameFloatCount: Int { channels * imgSize * imgSize }    // 15552
    static var inputFloatCount: Int { framesIn * frameFloatCount }      // 2_814_912
}
